src/View/SSViewer.php

```diff
@@ -719,177 +359,6 @@ public function process($item, $arguments = null, $inheritedScope = null)
-        $v = SSViewer::fromString($content);
-
-        if ($globalRequirements) {
-            $v->includeRequirements(false);
-        } else {
-            //nest a requirements backend for our template rendering
-            $origBackend = Requirements::backend();
-            Requirements::set_backend(Requirements_Backend::create());
-        }
-        try {
-            return $v->process($data, $arguments);
-        } finally {
-            if (!$globalRequirements) {
-                Requirements::set_backend($origBackend);
-            }
-        }
-    }
-
-    /**
-     * Parse given template contents
-     *
-     * @param string $content The template contents
-     * @param string $template The template file name
-     * @return string
-     * @deprecated 5.4.0 Will be replaced with SilverStripe\View\SSTemplateEngine::parseTemplateContent()
-     */
-    public function parseTemplateContent($content, $template = "")
-    {
-        Deprecation::noticeWithNoReplacment('5.4.0', 'Will be replaced with SilverStripe\View\SSTemplateEngine::parseTemplateContent()');
-        return $this->getParser()->compileString(
-            $content,
-            $template,
-            Director::isDev() && SSViewer::config()->uninherited('source_file_comments')
-        );
-    }
-
-    /**
-     * Returns the filenames of the template that will be rendered.  It is a map that may contain
-     * 'Content' & 'Layout', and will have to contain 'main'
-     *
-     * @return array
-     * @deprecated 5.4.0 Will be removed without equivalent functionality to replace it
-     */
-    public function templates()
-    {
-        Deprecation::noticeWithNoReplacment('5.4.0');
-        return array_merge(['main' => $this->chosen], $this->subTemplates);
-    }
-
-    /**
-     * @param string $type "Layout" or "main"
-     * @param string $file Full system path to the template file
-     * @deprecated 5.4.0 Will be removed without equivalent functionality to replace it
-     */
-    public function setTemplateFile($type, $file)
-    {
-        Deprecation::noticeWithNoReplacment('5.4.0');
-        if (!$type || $type == 'main') {
-            $this->chosen = $file;
-        } else {
-            $this->subTemplates[$type] = $file;
-        }
-    }
-
-    /**
-     * Return an appropriate base tag for the given template.
-     * It will be closed on an XHTML document, and unclosed on an HTML document.
-     *
-     * @param string $contentGeneratedSoFar The content of the template generated so far; it should contain
-     * the DOCTYPE declaration.
-     * @return string
-     * @deprecated 5.4.0 Use getBaseTag() instead
-     */
-    public static function get_base_tag($contentGeneratedSoFar)
-    {
-        Deprecation::notice('5.4.0', 'Use getBaseTag() instead');
-        // Is the document XHTML?
-        $isXhtml = preg_match('/<!DOCTYPE[^>]+xhtml/i', $contentGeneratedSoFar ?? '');
-        return static::getBaseTag($isXhtml);
-    }
```

Comment:
Replaced with the new implementation because all template engines should use this base tag, but not all rendering engines will be able to pass what has been rendered up to this point into the method.